### ISSUE TITLE
mongo-gorm stack overflows on startup

### ISSUE THREAD

**Issue by MitchJust** [micronaut-projects/micronaut-groovy#48]
To reproduce: (using micronaut 1.3.4)

`$ mn create-app --inplace --lang groovy --features mongo-gorm`

Add a simple entity class:

```groovy
@Entity
class SomeEntity {
    ObjectId id
    String something

    static constraints = {
        something nullable: false
    }
}
```

`$ docker run -d -p 27017:27017 --name mongo mongo:3.6`
`$ ./gradlew run`
```4:39:19.711 [main] ERROR io.micronaut.runtime.Micronaut - Error starting Micronaut server: Bean definition [org.grails.datastore.mapping.mongo.MongoDatastore] could not be loaded: Error instantiating bean of type [org.grails.datastore.mapping.mongo.MongoDatastore]: null
io.micronaut.context.exceptions.BeanInstantiationException: Bean definition [org.grails.datastore.mapping.mongo.MongoDatastore] could not be loaded: Error instantiating bean of type [org.grails.datastore.mapping.mongo.MongoDatastore]: null
	at io.micronaut.context.DefaultBeanContext.initializeContext(DefaultBeanContext.java:1359)
	at io.micronaut.context.DefaultApplicationContext.initializeContext(DefaultApplicationContext.java:236)
	at io.micronaut.context.DefaultBeanContext.readAllBeanDefinitionClasses(DefaultBeanContext.java:2547)
	at io.micronaut.context.DefaultBeanContext.start(DefaultBeanContext.java:204)
	at io.micronaut.context.DefaultApplicationContext.start(DefaultApplicationContext.java:188)
	at io.micronaut.runtime.Micronaut.start(Micronaut.java:64)
	at io.micronaut.runtime.Micronaut.run(Micronaut.java:294)
	at io.micronaut.runtime.Micronaut.run(Micronaut.java:280)
	at com.mitchjust.Application.main(Application.groovy:9)
Caused by: io.micronaut.context.exceptions.BeanInstantiationException: Error instantiating bean of type [org.grails.datastore.mapping.mongo.MongoDatastore]: null
	at io.micronaut.context.DefaultBeanContext.doCreateBean(DefaultBeanContext.java:1721)
	at io.micronaut.context.DefaultBeanContext.createAndRegisterSingletonInternal(DefaultBeanContext.java:2407)
Caused by: io.micronaut.context.exceptions.BeanInstantiationException: Error instantiating bean of type [org.grails.datastore.mapping.mongo.MongoDatastore]: null

	at io.micronaut.context.DefaultBeanContext.loadContextScopeBean(DefaultBeanContext.java:1994)
	at io.micronaut.context.DefaultBeanContext.initializeContext(DefaultBeanContext.java:1357)
	... 8 common frames omitted
Caused by: java.lang.StackOverflowError: null
at org.grails.datastore.mapping.config.ConfigurationBuilder.buildRecurse(ConfigurationBuilder.groovy:330)
        at org.grails.datastore.mapping.config.ConfigurationBuilder.buildRecurse(ConfigurationBuilder.groovy:187)
        at org.grails.datastore.mapping.config.ConfigurationBuilder.buildRecurse(ConfigurationBuilder.groovy:187)
        at org.grails.datastore.mapping.config.ConfigurationBuilder.buildRecurse(ConfigurationBuilder.groovy:187)
....
```

**Comment by sdelamo** [comment]
I pushed a sample app which reproduces the issue: 

https://github.com/grails-core-issues-forks/micronaut-projects-micronaut-groovy-issues-48

**Comment by sdelamo** [comment]
This seems to be an issue in GORM for Mongo DB:

https://github.com/grails/gorm-mongodb/issues/104

**Comment by sdelamo** [comment]
@hesara provided a workaround int the previously linked issue:

> Although not ideal as it disables the new auto-encryption functionality, a quick workaround for those who don't need it is to add 'autoEncryptionSettings' to ignored fields e.g. at application startup:

> Field field = ConfigurationBuilder.class.getDeclaredField('IGNORE_METHODS')
> field.setAccessible(true)
> ((Set) field.get(null)).add('autoEncryptionSettings')
> This at least enables the Micronaut application to start up.

If you setup this in your main class the app will start:

```
package example.micronaut

import io.micronaut.runtime.Micronaut
import groovy.transform.CompileStatic
import org.grails.datastore.mapping.config.ConfigurationBuilder

import java.lang.reflect.Field

@CompileStatic
class Application {
    static void main(String[] args) {
        Field field = ConfigurationBuilder.class.getDeclaredField('IGNORE_METHODS')
        field.setAccessible(true)
        ((Set) field.get(null)).add('autoEncryptionSettings')

        Micronaut.run(Application)
    }
}
```

**Comment by sdelamo** [comment]
Issue is resolved in the 7.0.5.BUILD-SNAPSHOT. you could set:

```groovy
repositories {
    ...
    maven { url "https://repo.grails.org/grails/core" }
}
....
dependencies {
    ....
    implementation "io.micronaut.configuration:micronaut-mongo-reactive:1.3.0"
   ....
    implementation("io.micronaut.configuration:micronaut-mongo-gorm") {
        exclude(group: "org.grails", module: 'grails-datastore-core')
    }
    implementation "org.grails:grails-datastore-core:7.0.5.BUILD-SNAPSHOT"
```

I have updated the sample app with the fix: 

https://github.com/grails-core-issues-forks/micronaut-projects-micronaut-groovy-issues-48
